app/src/sharedTest/java/org/oppia/app/topic/TopicFragmentTest.kt

```diff
@@ -0,0 +1,350 @@
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle(getResourceString(R.string.play))))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnPlayTab_showsPlayTabWithContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.play),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withText("First Story")).check(matches(isDisplayed()))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnTrainTab_showsTrainTabSelected() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.train),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle(getResourceString(R.string.train))))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnTrainTab_showsTrainTabWithContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.train),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withText("Master These Skills")).check(matches(isDisplayed()))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnReviewTab_showsReviewTabSelected() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.review),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle(getResourceString(R.string.review))))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnReviewTab_showsReviewTabWithContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.review),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        atPosition(R.id.review_skill_recycler_view, 0).also { withText("An important skill") },
+        isDescendantOfA(withId(R.id.review_skill_recycler_view))
+      ).also { withId(R.id.skill_name) })
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnOverviewTab_showsOverviewTabSelected() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.review),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        withText(R.string.overview),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle(getResourceString(R.string.overview))))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnOverviewTab_showsOverviewTabWithContentMatched() {
```

Comment:
Is this testing tab switching? If so, please indicate in the test name or remove the extra switching in the test, since the two don't match.

E.g. the test name could be: ``testTopicFragment_clickOnReviewTab_thenOverviewTab_showsOverviewTab``